vncviewer/vncviewer.cxx

```diff
@@ -503,6 +503,30 @@ static int mktunnel()
 }
 #endif /* !WIN32 */
 
+static int mainloop(const char* vncserver, network::Socket* sock)
+{
+  if(exitError != NULL) {
+    free(exitError);
+    exitError = NULL;
+  }
+
+  CConn *cc = new CConn(vncServerName, sock);
+
+  inMainloop = true;
+  exitMainloop = false;
+  while (!exitMainloop)
+    run_mainloop();
+  inMainloop = false;
+
+  delete cc;
+
+  if (exitError != NULL && alertOnFatalError)
```

Comment:
The description for `alertOnFatalError` I am aware of is: 
> Display a dialog with any fatal error before exiting.
So the dialog is still shown, just with the additional option to retry. If the option is off no dialog is shown and no retry is done.

Reply:
Yeah, thats fine if the "Retry" only applies to the initial connection.

Reply:
Since you explained above that this does apply to a successful connection I think this feature might be more important than showing an error before exiting, for some users.

I think we should add info about the reconnect-feature to the description of that parameter at the very least. Perhaps even split this behavior to it's own parameter.